sources/packages/backend/libs/services/src/confirmation-of-enrollment/confirmation-of-enrollment.service.ts

```diff
@@ -111,4 +141,532 @@ export class ConfirmationOfEnrollmentService {
+      tuitionRemittanceAmount,
+      disbursementSchedule.id,
+    );
+
+    await this.updateDisbursementAndApplicationCOEApproval(
+      disbursementScheduleId,
+      auditUserId,
+      disbursementSchedule.studentAssessment.application.id,
+      disbursementSchedule.studentAssessment.application.applicationStatus,
+      tuitionRemittanceAmount,
+      options?.enrolmentConfirmationDate,
+    );
+  }
+
+  /**
+   * Decline the enrolment of an application.
+   * @param disbursementScheduleId disbursement schedule id.
+   * @param auditUserId user who confirms enrollment.
+   * @param declineReason reason for declining the enrolment.
+   * @param options decline enrolment options.
+   * - `locationId` location id of the application.
+   * - `applicationNumber` application number of the enrolment.
+   */
+  async declineEnrolment(
+    disbursementScheduleId: number,
+    auditUserId: number,
+    declineReason: { coeDenyReasonId: number; otherReasonDesc?: string },
+    options: {
+      locationId?: number;
+      applicationNumber?: string;
+    },
+  ): Promise<void> {
+    const disbursementSchedule =
+      await this.getDisbursementAndApplicationSummary(
+        disbursementScheduleId,
+        options?.locationId,
+      );
+
+    if (!disbursementSchedule) {
+      throw new CustomNamedError("Enrolment not found.", ENROLMENT_NOT_FOUND);
+    }
+
+    const application = disbursementSchedule.studentAssessment.application;
+
+    if (
+      options?.applicationNumber &&
+      options?.applicationNumber !== application.applicationNumber
+    ) {
+      throw new CustomNamedError(
+        "Enrolment for the given application not found.",
+        ENROLMENT_NOT_FOUND,
+      );
+    }
+
+    if (disbursementSchedule.coeStatus !== COEStatus.required) {
+      throw new CustomNamedError(
+        "Enrolment already completed and can neither be confirmed nor declined.",
+        ENROLMENT_ALREADY_COMPLETED,
+      );
+    }
+
+    if (
+      ![ApplicationStatus.Enrolment, ApplicationStatus.Completed].includes(
+        disbursementSchedule.studentAssessment.application.applicationStatus,
+      )
+    ) {
+      throw new CustomNamedError(
+        "Enrolment cannot be declined as application is not in a valid status.",
+        ENROLMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE,
+      );
+    }
+
+    await this.updateCOEToDeclined(disbursementSchedule.id, auditUserId, {
+      coeDeniedReasonId: declineReason.coeDenyReasonId,
+      otherReasonDesc: declineReason.otherReasonDesc,
+    });
+  }
+
+  /**
+   * Validate Institution Users to request tuition remittance at the time
+   * of confirming enrolment, not to exceed the lesser than both
+   * (Actual tuition + Program related costs) and (Canada grants + Canada Loan + BC Loan).
+   * @param tuitionRemittanceAmount tuition remittance submitted by institution.
+   * @param disbursementScheduleId disbursement schedule id.
+   * @throws UnprocessableEntityException.
```

Comment:
this function is not throwing `UnprocessableEntityException` here, right? its in the controller